presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueryFramework.java

```diff
@@ -70,7 +70,7 @@ protected AbstractTestQueryFramework(QueryRunnerSupplier supplier)
     public void init()
             throws Exception
     {
-        queryRunner = queryRunnerSupplier.get();
+        queryRunner = queryRunner != null ? queryRunner : queryRunnerSupplier.get();
```

Comment:
This seems wrong. The `@BeforeClass` method should never be called more than once.